dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -560,6 +563,18 @@ void TypeLookupService::complete_to_dynamic_i(DynamicTypeImpl* dt,
   case TK_ENUM:
     td->kind(TK_ENUM);
     td->name(cto.enumerated_type.header.detail.type_name.c_str());
+    if (cto.enumerated_type.enum_flags & IS_FINAL) {
+      td->extensibility_kind(DDS::FINAL);
+    } else if (cto.enumerated_type.enum_flags & IS_APPENDABLE) {
+      td->extensibility_kind(DDS::APPENDABLE);
+    } else {
+      if (DCPS::log_level >= DCPS::LogLevel::Warning) {
+        ACE_ERROR((LM_WARNING,
+                   "(%P|%t) WARNING: TypeLookupService::complete_to_dynamic_i: "
+                   "Invalid extensibility kind in TK_ENUM, using default\n"));
+      }
+      td->extensibility_kind(default_enum_extensibility_);
```

Comment:
Follow up on my other comment on `default_enum_extensibility` function in `TypeLookupService`. So this happens when we receive a TypeObject from an old system but after the type compatibility check is already done using the received TypeObject?

It looks like we're providing compatibility with old systems only when the enum type isn't annotated and `opendds_idl` in the new system is called with `--default-enum-extensibility-zero`. If the enum type is annotated, I think it won't work.

Reply:
You are right.  We need to reconsider the design here.